app/routes/quotes/components/AddQuote.tsx

```diff
@@ -30,7 +30,23 @@ const validate = createValidator({
 });
 
 const AddQuote = ({ addQuotes, actionGrant }: Props) => {
-  const onSubmit = withSubmissionErrorFinalForm(addQuotes);
+  const removeUnnecessaryDash = (source: string) => {
+    if (source == undefined) return undefined;
+
+    const dashIndex = source.indexOf('-');
+    if (source.slice(0, dashIndex).match(/^ *$/)) {
+      source = source.slice(dashIndex + 1);
+    }
+
+    return source;
+  };
+
+  const onSubmit = (quote: { text: string; source: string }) => {
+    return withSubmissionErrorFinalForm(addQuotes)({
+      text: quote.text,
+      source: removeUnnecessaryDash(quote.source),
+    });
+  };
```

Comment:
When all the function does is return something, you may use an implicit return. Not necessary, rather just nice to know.

```suggestion
  const onSubmit = (quote: { text: string; source: string }) => (
    withSubmissionErrorFinalForm(addQuotes)({
      text: quote.text,
      source: removeUnnecessaryDash(quote.source),
    })
  );
```